punishments.js

```diff
@@ -746,6 +789,34 @@ Punishments.roomUnblacklistAll = function (room) {
 	return unblacklisted;
 };
 
+/**
+ * @param {string} ip
+ * @param {string} note
+ */
+Punishments.addSharedIp = function (ip, note) {
+	Punishments.sharedIps.set(ip, note);
+	Punishments.appendSharedIp(ip, note);
+
+	Users.users.forEach(user => {
+		if (user.locked && user.locked !== user.userid && user.latestIp === ip) {
```

Comment:
>latestIp

This needs to go through the user's entire IP list.

Also, the shared IP needs to be removed from `Punishments.ips`